### PR TITLE
[dagster-dbt] Upgrade dagster-dbt to be compatible with dbt 1.5.x

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

Fixes: https://github.com/dagster-io/dagster/issues/13898

The changes:

- `--no-use-color` is deprecated (`--no-use-colors` is allowed on all versions 1.0+)
- The dbt selection string parsing logic relies on global state being set (I know, I know...), and the way that this global state is set has been changed (previously, they were attributes directly on the module, now they're all wrapped into a single GLOBAL_FLAGS object)
- Nasty hack to deal with dbt-core shifting from `argparse` to `click`. We were using the argument parsing functionality to parse dbt Cloud job commands, so I found a somewhat similar entrypoint to work with.
- Ensure that the current working directory when executing dbt commands is the project directory. This handles the issue linked below, where dbt now writes output files to the cwd rather than the project directory. This is not an issue if the cwd *is* the project directory ;)
- Update some tests to be a bit more flexible to subtly different output formats (generally this is just testing stuff in the DbtCliOutput object, which we don't particularly expect or need to be very stable)
- Some errors got reworded, so tests needed to be updated

In general, the `select_unique_ids_from_manifest` function access a lot of things it potentially shouldn't touch from the dbt internals, and there's no guarantee similar issues don't crop up in the future. The hope is that there is eventually a dbt-native python entrypoint for this functionality, at which point in time we can remove all this ugly stuff.

## How I Tested These Changes

Added a dbt_15X tox env, which is pinned to the specific rcs that were available to me. Once the real `dbt 1.5.x` comes out, this should be unpinned

### CODE REVIEW

**Comment by OwenKephart** [comment]
Currently blocked on: https://github.com/dbt-labs/dbt-core/issues/7465, although should be possible to work around

File: python_modules/libraries/dagster-dbt/dagster_dbt/cloud/asset_defs.py [hunk 1]
```diff
@@ -103,6 +102,8 @@ def build_definitions(
 
     @staticmethod
     def parse_dbt_command(dbt_command: str) -> Namespace:
+        from dbt.main import parse_args as dbt_parse_args
```
[the remark this answers is not in the document]
Comment (OwenKephart) [reply python_modules/libraries/dagster-dbt/dagster_dbt/cloud/asset_defs.py:105]: added comment, but I think we need to figure out a solution here for the dbt-core>=1.5.0 case

File: python_modules/libraries/dagster-dbt/dagster_dbt/utils.py [hunk 2]
```diff
@@ -300,7 +300,13 @@ def __getattr__(self, item):
...
 
     # create a parsed selection from the select string
-    flags.INDIRECT_SELECTION = IndirectSelection.Eager
+    try:
+        from dbt.flags import GLOBAL_FLAGS
+
+        setattr(GLOBAL_FLAGS, "INDIRECT_SELECTION", IndirectSelection.Eager)
+    except ImportError:
+        # dbt < 1.5.0 compat
+        flags.INDIRECT_SELECTION = IndirectSelection.Eager
```
[the remark this answers is not in the document]
Comment (OwenKephart) [reply python_modules/libraries/dagster-dbt/dagster_dbt/utils.py]: nice

**Comment by OwenKephart** [comment]
Ok avoided the `dbt-core` bug for now, so that's no longer blocking.

**Comment by tha23rd** [comment]
@OwenKephart 

Not sure if this is beneficial for this PR at all, but wanted to put this on your radar (coming out in v1.5): https://docs.getdbt.com/reference/programmatic-invocations

I know dagster-dbt does a lot of stuff behind the scenes with the CLI, so this might help clean things up a bit

**Comment by rexledesma** [comment]
> @OwenKephart
> 
> Not sure if this is beneficial for this PR at all, but wanted to put this on your radar (coming out in v1.5): https://docs.getdbt.com/reference/programmatic-invocations
> 
> I know dagster-dbt does a lot of stuff behind the scenes with the CLI, so this might help clean things up a bit

@tha23rd Owen mentioned this internally to our team! Yeah this will be such a great QOL cleanup for our integration.

**Review by rexledesma — APPROVED** [review]

**Comment by rexledesma** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13907** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13907" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13907?utm_source=stack-comment).

**Pull request merged**